fabric-item-api-v1/src/main/java/net/fabricmc/fabric/impl/item/ShieldRegistryImpl.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.item;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemConvertible;
+
+import net.fabricmc.fabric.api.item.v1.ShieldRegistry;
+
+public class ShieldRegistryImpl {
+	private static final Map<Item, ShieldRegistry.Entry> registeredItemEntries = new HashMap<>();
```

Comment:
I suggest using object 2 int map from fastutil instead and set the default return value for that map to `0`. Then remove that entry class

Reply:
My reasoning behind an Entry class was avoiding the possibility of an NPE on getAxeDisableDuration for an unregistered item. It also leaves options open for adding more features to the registry in the future like custom sounds or others when a combat update finally happens.

Reply:
I would consider mixing into `Item` and storing the entry there. That would reduce the amount of `Map.get` calls as you could just cast and get the entry.

I'd probably just have a static method in api class to get the cooldown time from an Item and then if in the future we add new features, just add a new method in api to get the value.

Reply:
I tried storing the entry on the Item and I feel like that works really well. I can already see a usecase of sticking with the entry since in the latest combat update shield only block up to 5 damage, which could be customized with the registry. (if this feature makes it to a major minecraft version)